panic in buildengine if time fails to build

If time has a build error, the build engine will panic when attempting to build time

I added this code to `time.go` (notice the SumType is missing `//ftl:enum export`)
```go
type SumType interface {
	tag()
}

type A string

func (A) tag() {}

type B []string

func (B) tag() {}

//ftl:ingress http GET /typeenum
func TypeEnum(ctx context.Context, req builtin.HttpRequest[SumType]) (builtin.HttpResponse[SumType, string], error) {
	return builtin.HttpResponse[SumType, string]{Body: ftl.Some(req.Body)}, nil
}
```

```bash
ftl dev --recreate
info: Starting FTL with 1 controller(s)
info:controller0: Web console available at: http://localhost:8892
info:controller0: HTTP ingress server listening on: http://localhost:8891
info: Controller started in 1.007478292s
info: FTL startup command ⚡️
warn: module: builtin
warn: mc: time
info:time: Building module
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x48 pc=0x104eda79c]

goroutine 243 [running]:
github.com/TBD54566975/ftl/go-runtime/compile.GenerateStubsForModules({0x105bcb4d0, 0x140000ea690}, {0x1400066a780, 0x19}, {0x14000ec2180, 0x1, 0x105920480?}, 0x14000bc4580)
	/Users/wesbillman/dev/ftl/go-runtime/compile/build.go:245 +0x77c
github.com/TBD54566975/ftl/buildengine.generateGoStubs({0x105bcb4d0, 0x140000ea690}, {0x1400066a780, 0x19}, {0x14001c081f0, 0x2, 0x2}, {0x14000ec2180, 0x1, 0x1})
	/Users/wesbillman/dev/ftl/buildengine/stubs.go:21 +0xa4
github.com/TBD54566975/ftl/buildengine.GenerateStubs(...)
	/Users/wesbillman/dev/ftl/buildengine/stubs.go:16
github.com/TBD54566975/ftl/buildengine.(*Engine).buildWithCallback(0x140003f9c00, {0x105bcb4d0, 0x140000ea690}, 0x14000668ca0, {0x14000668c80, 0x2, 0x2})
	/Users/wesbillman/dev/ftl/buildengine/engine.go:582 +0x500
github.com/TBD54566975/ftl/buildengine.(*Engine).BuildAndDeploy.func1()
	/Users/wesbillman/dev/ftl/buildengine/engine.go:506 +0xe0
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/Users/wesbillman/go/pkg/mod/golang.org/x/sync@v0.7.0/errgroup/errgroup.go:78 +0x58
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 76
	/Users/wesbillman/go/pkg/mod/golang.org/x/sync@v0.7.0/errgroup/errgroup.go:75 +0x98
```

--- a/buildengine/build.go
+++ b/buildengine/build.go
@@ -41,6 +41,9 @@ func buildModule(ctx context.Context, projectRootDir string, sch *schema.Schema,
 	}
 
 	logger.Infof("Building module")
+
+	startTime := time.Now()
+
 	switch module.Config.Language {
 	case "go":
 		err = buildGoModule(ctx, projectRootDir, sch, module, filesTransaction)
@@ -69,6 +72,8 @@ func buildModule(ctx context.Context, projectRootDir string, sch *schema.Schema,
 	if len(errs) > 0 {
 		return errors.Join(errs...)
 	}
+
+	logger.Infof("Module built (%.2fs)", time.Since(startTime).Seconds())
 
 	return nil
 }

--- a/buildengine/engine.go
+++ b/buildengine/engine.go
@@ -729,7 +729,10 @@ func (e *Engine) gatherSchemas(
 		latestModule = moduleMeta{module: module}
 	}
 	for _, dep := range latestModule.module.Dependencies {
-		out[dep] = moduleSchemas[dep]
+		if moduleSchemas[dep] != nil {
+			out[dep] = moduleSchemas[dep]
+		}
+
 		if dep != "builtin" {
 			depModule, ok := e.moduleMetas.Load(dep)
 			// TODO: should we be gathering schemas from dependencies without a module?

